File: 0_Array/1_Searching/CPP/linear_search.cpp
#include <iostream>
using namespace std;

int linear_search(int *arr, int n, int key)
{
    for (int i = 0; i < n; i++)
    {
        if (arr[i] == key)
            return i;
    }
    return -1;
}

int main()
{
    int arr[] = {12, 34, 23, 56, 76, 34, 45, 7};
    int n = sizeof(arr) / sizeof(int);
    int key;
    cin >> key;
    int index = linear_search(arr, n, key);
    if (index != -1)
    {
        cout << key << " is present at index " << index << endl;
    }
    else
        cout << key << " NOT found!" << endl;
    return 0;
}

File: 0_Array/1_Searching/Python/linear_search.py
def linear_search(arr, key):
    print(len(arr))
    for i in range(len(arr)):
        if arr[i] == key:
            return i
    return -1


arr = [12, 43, 45, 65, 37, 45]
key = int(input("Enter key: "))
index = linear_search(arr, key)
if (index != -1):
    print(key, "is at position", index)
else:
    print(key, "NOT found!")
